src/js/schemas/service-schema/EnvironmentVariables.js

```diff
@@ -21,12 +21,20 @@ let EnvironmentVariables = {
         if (variableMap == null) {
           return [];
         }
-        return Object.keys(variableMap).map(function (key) {
-          return Hooks.applyFilter('variablesGetter', {
+
+        return Object.keys(variableMap).reduce(function (memo, key) {
+          if ((key == null || key === 'undefined')
```

Comment:
We only run the risk of filtering out a variable where both the key AND value is `undefined`, `"undefined"` or `null`. So I think its safe.

Kenny's right, `typeof key === 'undefined'` won't work reliably here because it gets converted to a String.